### DISCUSSION TITLE
How to login to nhost within a next.js API route?

### DISCUSSION THREAD

**Discussion by WouterrV** [nhost/nhost#1429] [Q&A]
Started using nhost a few days ago and really enjoying it so far, but I can't figure out how to make properly authorized requests from the server side (which is what API routes are in Next.js, these don't run in the browser).

I'm doing `import { NhostClient } from '@nhost/nhost-js'` to then create a client.

Doing 
```
await nhost.auth.signIn({
    email: 'xxxxx@hotmail.com',
     password: 'xxxx',
})
```

works just fine, but I feel it's more appropriate to do:
`nhost.functions.setAccessToken(nhostToken)`

Which doesn't seem to work even though I'm passing the token that the frontend uses. Thoughts?

**Comment by stefan-girlich** [comment]
@WouterrV I'm almost [in the same situation as you](https://github.com/nhost/nhost/discussions/2932), just using Next.js's new route handlers which succeed API routes.

Did you ever find a solution to your problem? I can access the cookies containing the refresh token, but I yet have to find a way to pass it to the server-side nhost client.
